packages/@dcl/ecs/src/engine/types.ts

```diff
@@ -238,6 +239,26 @@ export interface IEngine {
     ...components: T
   ): Iterable<[Entity, ...ReadonlyComponentSchema<T>]>
 
+  /**
+   * Get an iterator of entities that follow a tree structure for a component
+   * @param entity - the root entity of the tree
+   * @param component - the parenting component to filter by
+   * @returns An iterator of an array as [entity, entity2, ...]
+   *
+   * Example:
+   * ```ts
+   * const TreeComponent = engine.defineComponent('custom::TreeComponent', {
+   *    label: Schemas.String,
+   *    parent: Schemas.Entity
+   * })
+   *
+   * for (const entity of getComponentEntityTree(entity, TreeComponent)) {
+   *    // entity in the tree
+   * }
+   * ```
+   */
+  getComponentEntityTree<T>(entity: Entity, component: ComponentDefinition<T & { parent?: Entity }>): Generator<Entity>
```

Comment:
I think this should be a helper function outside "engine", could you extract it?